### ISSUE TITLE
Integrate User/Peer Availability with Server

### ISSUE THREAD

**Issue by MarkBelonogoff** [hkhosrav/RiPPLE-Core#111]
Implement the Availability API on the server and update the client to request following availability actions:

- The current availability of the user for the course
- The popularity of the times slots for the course (Excluding the current user?)
- Insert a new availability for the user
- Remove a current time slot the user no longer wants

**Comment by MarkBelonogoff** [comment]
I'm trying to update the availabilities by creating a function that would call the Promise at
`PeerView.vue:114` but I'm not sure how to listen for the event as it is emitted at
`components/UserContainer.vue:93`.

How should I implement this?
Thanks

**Comment by CaptainAchilles** [comment]
Hey @MarkBelonogoff,

I had this issue early on in the project and solved it by creating an eventing  system (named `Fetcher`). If you need data which is *always* up to date (eg. Topics, User Information) then I would recommend using the `Fetcher` subscription service.

If you have data which is very reactive (such as question searching, or competency filtering); then I would recommend manually managing the state in your components.

---
If you look at the `UserBadge` component, you will see an example of how to use the `Fetcher`. Essentially you _subscribe_ to a function reference in your component, and bind that to an internal component variable, then whenever a global refresh is triggered (such as a course change), all subscribed functions will be re-called and the data will be refreshed.

**Comment by MarkBelonogoff** [comment]
I have a bug where a user-selected availability checkbox remains checked for some time after switching the user course. This happens even if the user is unavailable for that selected time slot for the newly selected course.

I have tried adding the course code to the checkbox id but the bug persists even with the id name change. 

@CaptainAchilles  How do I connect the Fetch function to the v-model for these checkboxes to see if that will force a refresh?

**Comment by CaptainAchilles** [comment]
Hey @MarkBelonogoff 

It seems that your issue has been fixed at https://khosravi.uqcloud.net/ripple/#/view/peers.
Admittedly; I did change `PeerView.vue:178` since it was giving compile errors with the introduction of generics to the `apiFetch` function.

```typescript
    changeAvailability(day, time) {
        AvailabilityService.updateUserAvailability(day, time)
        .then(x => AvailabilityService.getCourseAvailability())
        .then(this.updateCourseAvailability);
    }
```

Can you confirm that its fixed and we can close the ticket given that #186 has also been merged.
